src/bijectors/named_bijector.jl

```diff
@@ -0,0 +1,231 @@
+abstract type AbstractNamedBijector <: AbstractBijector end
+
+forward(b::AbstractBijector, x) = (rv = b(x), logabsdetjac = logabsdetjac(b, x))
+
+#######################
+### `NamedBijector` ###
+#######################
+"""
+    NamedBijector <: AbstractNamedBijector
+
+Wraps a `NamedTuple` of key -> `Bijector` pairs, implementing evaluation, inversion, etc.
+
+# Examples
+```julia-repl
+julia> using Bijectors: NamedBijector, Scale, Exp
+
+julia> b = NamedBijector((a = Scale(2.0), b = Exp()));
+
+julia> x = (a = 1., b = 0., c = 42.);
+
+julia> b(x)
+(a = 2.0, b = 1.0, c = 42.0)
+
+julia> (a = 2 * x.a, b = exp(x.b), c = x.c)
+(a = 2.0, b = 1.0, c = 42.0)
+```
+"""
+struct NamedBijector{names, Bs<:NamedTuple{names}} <: AbstractNamedBijector
+    bs::Bs
+end
+
+@inline names_to_bijectors(b::NamedBijector) = b.bs
+
+@generated function (b::NamedBijector{names})(x::NamedTuple) where {names}
+    return :(merge(x, ($([:($n = b.bs.$n(x.$n)) for n in names]...), )))
+end
+
+@generated function Base.inv(b::NamedBijector{names}) where {names}
+    return :(NamedBijector(($([:($n = inv(b.bs.$n)) for n in names]...), )))
+end
```

Comment:
Is the generated function actually needed here? Would be interesting to benchmark if it is faster than just
```suggestion
Base.inv(b::NamedBijector) = NamedBijector(map(inv, b.bs))
```
or
```suggestion
function Base.inv(b::NamedBijector{names}) where {names}
	bijectors = map(inv, b.bs)
	return NamedBijector{names,typeof(bijectors)}(bijectors)
end
```

Reply:
That's one thing, but there's also the issue of type-stability. `map(inv, b.bs)` will fail if `bs` is large :confused: